packages/devextreme/testing/testcafe/update_failed_etalons.mjs

```diff
@@ -0,0 +1,91 @@
+import {
+  readdirSync, lstatSync, existsSync, copyFileSync,
+} from 'fs';
+import { join, resolve } from 'path';
+import { argv } from 'process';
+
+const PATH_TO_CURRENT_ETALONS = argv[2];
+const PATH_TO_TESTCAFE_TESTS = './testing/testcafe/tests';
+const NEW_ETALON_FILE_NAME_PATTERN = /^(?!.*(?:_mask|_diff|_etalon)\.png$).*\.png$/i;
+const ETALONS_FOLDER_NAME = 'etalons';
+
+const getAllEtalonsNames = () => {
+  const allFileNames = readdirSync(PATH_TO_CURRENT_ETALONS);
+  const newEtalons = allFileNames.filter((name) => NEW_ETALON_FILE_NAME_PATTERN.test(name));
+
+  return newEtalons;
```

Comment:
I suggest u creating a map here instead of a plain array for an access with O(1) time.
It will speed up processing because we don't need to make linear operations like `splice` and `indexOf`

Reply:
fixed